src/screen.cpp

```diff
@@ -198,70 +202,77 @@ static bool is_csi_style_escape_seq(const wchar_t *code, size_t *resulting_lengt
+                return true;
             }
         }
+    }
 
-        // Detect these semi-common terminfo escapes without any parameter values, all of which
-        // don't move the cursor.
-        char *const esc2[] = {enter_bold_mode,        exit_attribute_mode,   enter_underline_mode,
-                              exit_underline_mode,    enter_standout_mode,   exit_standout_mode,
-                              flash_screen,           enter_subscript_mode,  exit_subscript_mode,
-                              enter_superscript_mode, exit_superscript_mode, enter_blink_mode,
-                              enter_italics_mode,     exit_italics_mode,     enter_reverse_mode,
-                              enter_shadow_mode,      exit_shadow_mode,      enter_standout_mode,
-                              exit_standout_mode,     enter_secure_mode,     enter_dim_mode,
-                              enter_blink_mode,       enter_protected_mode,  enter_alt_charset_mode,
-                              exit_alt_charset_mode};
-
-        for (size_t p = 0; p < sizeof esc2 / sizeof *esc2 && !found; p++) {
-            if (!esc2[p]) continue;
-            // Test both padded and unpadded version, just to be safe. Most versions of tparm don't
-            // actually seem to do anything these days.
-
-            size_t len = maxi(try_sequence(tparm(esc2[p]), code), try_sequence(esc2[p], code));
-            if (len) {
-                resulting_length = len;
-                found = true;
-            }
+    return false;
+}
+
+// Detect whether the escape sequence sets one of the terminal attributes that affects how text is
+// displayed other than the color.
+static bool is_visual_escape_seq(const wchar_t *code, size_t *resulting_length) {
+    if (!cur_term) return false;
+    char *const esc2[] = {enter_bold_mode,        exit_attribute_mode,   enter_underline_mode,
+                            exit_underline_mode,    enter_standout_mode,   exit_standout_mode,
+                            flash_screen,           enter_subscript_mode,  exit_subscript_mode,
+                            enter_superscript_mode, exit_superscript_mode, enter_blink_mode,
+                            enter_italics_mode,     exit_italics_mode,     enter_reverse_mode,
+                            enter_shadow_mode,      exit_shadow_mode,      enter_standout_mode,
+                            exit_standout_mode,     enter_secure_mode,     enter_dim_mode,
+                            enter_blink_mode,       enter_protected_mode,  enter_alt_charset_mode,
+                            exit_alt_charset_mode};
+
+    for (size_t p = 0; p < sizeof esc2 / sizeof *esc2; p++) {
+        if (!esc2[p]) continue;
+        // Test both padded and unpadded version, just to be safe. Most versions of tparm don't
+        // actually seem to do anything these days.
+        size_t esc_seq_len = maxi(try_sequence(tparm(esc2[p]), code), try_sequence(esc2[p], code));
```

Comment:
Not related.to any changes you made. But wonder what we have `maxi()` when we could just use `std::max`?